fbw-a380x/src/wasm/systems/a380_systems/src/hydraulic/autobrakes.rs

```diff
@@ -541,5 +604,777 @@ impl SimulationElement for A380AutobrakeController {
+    }
+
+    fn accel_to_reach_to_decelerate(&self) -> Acceleration {
+        self.desired_deceleration * Self::DECEL_RATIO_TO_REACH_TO_START_DECEL
+    }
+
+    fn accel_during_rot_opti(&self) -> Acceleration {
+        Acceleration::new::<meter_per_second_squared>(Self::ROT_OPTIMIZATION_TARGET_DECEL_M_S_2)
+    }
+
+    // Safety margin gives a dynamic ratio on targeted decel based on remaining distance
+    fn safety_margin(&self) -> f64 {
+        match self.state {
+            BTVState::Decel | BTVState::EndOfBraking | BTVState::_OutOfDecelRange => {
+                let ratio_of_decel_distance =
+                    self.braking_distance_remaining() / self.distance_remaining_at_decel_activation;
+
+                (1. + (ratio_of_decel_distance.get::<ratio>().sqrt()
+                    * Self::DECEL_SAFETY_MARGIN_SHAPING_FACTOR))
+                    .max(Self::MIN_DECEL_SAFETY_MARGIN_RATIO)
+                    .min(Self::MAX_DECEL_SAFETY_MARGIN_RATIO)
+            }
+
+            BTVState::Disabled | BTVState::Armed | BTVState::RotOptimization => {
+                Self::MAX_DECEL_SAFETY_MARGIN_RATIO
+            }
+        }
+    }
+
+    fn update_state(&mut self, ground_speed: Velocity) -> BTVState {
+        match self.state {
+            BTVState::Armed => {
+                if self.spoilers_active {
+                    self.update_desired_btv_deceleration();
+                    BTVState::RotOptimization
+                } else if !self.arming_authorized() {
+                    BTVState::Disabled
+                } else {
+                    self.state
+                }
+            }
+            BTVState::RotOptimization => {
+                let accel_min = self.accel_to_reach_to_decelerate();
+
+                if self.deceleration_request < accel_min {
+                    self.distance_remaining_at_decel_activation = self.braking_distance_remaining();
+                    self.end_of_decel_acceleration = self.deceleration_request;
+                    BTVState::Decel
+                } else {
+                    self.state
+                }
+            }
+            BTVState::Decel => {
+                if self.final_distance_remaining.get::<meter>() < Self::DISTANCE_TO_RELEASE_BTV_M
+                    || ground_speed.get::<meter_per_second>()
+                        <= Self::TARGET_SPEED_TO_RELEASE_BTV_M_S
+                {
+                    self.end_of_decel_acceleration = self.deceleration_request;
+                    BTVState::EndOfBraking
+                } else {
+                    BTVState::Decel
+                }
+            }
+            BTVState::EndOfBraking => {
+                if ground_speed.get::<meter_per_second>() <= Self::TARGET_SPEED_TO_RELEASE_BTV_M_S {
+                    self.disarm();
+                    BTVState::Disabled
+                } else {
+                    BTVState::EndOfBraking
+                }
+            }
+
+            _ => self.state,
```

Comment:
This is against the coding guidelines: https://github.com/flybywiresim/aircraft/blob/master/fbw-common/src/wasm/systems/guidelines.md#explicit-match-pattern-declaration